hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +62,106 @@
     }
 
     private long countMapWithWANReplication() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getWanReplicationRef() != null).count();
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static class LatencyInfo {
+        private Long totalLatency;
+        private Long operationCount;
+
+        LatencyInfo(Long totalLatency, Long operationCount) {
+            this.totalLatency = totalLatency;
+            this.operationCount = operationCount;
+        }
+
+        void add(Long totalLatency, Long operationCount) {
+            this.totalLatency += totalLatency;
+            this.operationCount += operationCount;
+        }
+
+        Long calculateAverage() {
+            return this.operationCount == 0 ? -1 : (this.totalLatency / this.operationCount);
+        }
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToGetLatency() {
+        return LocalMapStats::getTotalGetLatency;
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToGetOperationCount() {
+        return LocalMapStats::getGetOperationCount;
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToPutLatency() {
+        return LocalMapStats::getTotalPutLatency;
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToPutOperationCount() {
+        return LocalMapStats::getPutOperationCount;
+    }
+
+    private long mapOperationLatency(Node node, Predicate<MapConfig> predicate,
+                                     Function<? super LocalMapStats, ? extends Long> latencyMap,
+                                     Function<? super LocalMapStats, ? extends Long> operationMap) {
```

Comment:
3 notes on this:
  - the formal parameter names aren't descriptive. These are not maps. These are functions (used for mapping). I suggest something like `totalLatencyExtractor` , `totalLatencyProvider` , `totalLatencySupplier`, or even `totalLatencyFn` but not `*Map`, please. Also the second param is a function that returns the *operation count*. So it should be `operationCountProvider` or similar.
  - in case of external (public) APIs it is very wise not to restrict the generic type params to a single type (and use `extends` and `super` like you do here), but since it is a very small-scoped internal API, we are ok with `Function<LocalMapStats, Long>`. 
  - re-iterating the previous one, the problem with the use of primitive wrappers is that they come with a hell of implicit auto-boxing and unboxing. Since we still want to use the stream API with primitives, Java offers specialized function types for working with them. In this case, the appropriate specialized function type is `ToLongFunction` (which maps an object to a `long`). So to sum up, please use `ToLongFunction<LocalMapStats>` instead of `Function<? super LocalMapStats, ? extends Long>`.